src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -172,101 +181,127 @@ public boolean executeTask(
+                              getTargetAddressesToCompile(ModuleManager.getInstance(project).getModules()),
+                              Arrays.asList("export-classpath", "compile"));
   }
 
-  public Pair<Boolean, Optional<String>> executeTask(@NotNull Module[] modules) {
+  public PantsExecuteTaskResult executeCompileTask(@NotNull Module[] modules) {
     if (modules.length == 0) {
-      return Pair.create(false, Optional.empty());
+      return new PantsExecuteTaskResult(false, Optional.empty());
     }
-    return executeTask(modules[0].getProject(), getTargetAddressesToCompile(modules), false);
+    return executeCompileTask(modules[0].getProject(),
+                              getTargetAddressesToCompile(modules),
+                              Arrays.asList("export-classpath", "compile"));
   }
 
-  /**
-   * @param currentProject:           current project
-   * @param targetAddressesToCompile: set of target addresses to compile
-   * @param useCleanAll:              whether to use clean-all first in Pants
-   * @return whether the execution is successful, additional message along with the execution
-   * in a Pair object.
-   */
-  public Pair<Boolean, Optional<String>> executeTask(Project currentProject, Set<String> targetAddressesToCompile, boolean useCleanAll) {
+  @NotNull
+  public PantsExecuteTaskResult executeCompileTask(@NotNull Project currentProject, @NotNull Set<String> targetAddressesToCompile, @NotNull List<String> tasks) {
+
+    final String title = "Compile message";
+
     // If project has not changed since last Compile, return immediately.
     if (!FileChangeTracker.shouldRecompileThenReset(currentProject, targetAddressesToCompile)) {
       PantsExternalMetricsListenerManager.getInstance().logIsPantsNoopCompile(true);
-      Notification start = new Notification(PantsConstants.PANTS, "Pants Compile", "Already up to date.\n", NotificationType.INFORMATION);
-      Notifications.Bus.notify(start);
-      return Pair.create(true, Optional.of(PantsConstants.NOOP_COMPILE));
+      notify(title, "Already up to date.", NotificationType.INFORMATION);
+      return new PantsExecuteTaskResult(true, Optional.of(PantsConstants.NOOP_COMPILE));
     }
 
-    prepareIDE(currentProject);
-    if (targetAddressesToCompile.isEmpty()) {
-      showPantsMakeTaskMessage("No target found in configuration.\n", ConsoleViewContentType.SYSTEM_OUTPUT, currentProject);
-      return Pair.create(true, Optional.empty());
-    }
-
-    Optional<VirtualFile> pantsExecutable = PantsUtil.findPantsExecutable(currentProject);
-    if (!pantsExecutable.isPresent()) {
-      return Pair.create(
-        false,
-        Optional.of(
-          PantsBundle.message("pants.error.no.pants.executable.by.path", currentProject.getProjectFilePath())
-        )
-      );
-    }
-    final GeneralCommandLine commandLine = PantsUtil.defaultCommandLine(pantsExecutable.get().getPath());
+    List<String> pantsCmdLine = new ArrayList<>();
```

Comment:
nit: rename var and use simpler api for array, `List<String> pantsCmdArgs = Lists.newArrayList()`

Reply:
Done.